sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py

```diff
@@ -36,20 +36,23 @@ class SchemaRegistryAvroSerializer(object):
     SchemaRegistryAvroSerializer provides the ability to serialize and deserialize data according
     to the given avro schema. It would automatically register, get and cache the schema.
 
-    :param client: The schema registry client
+    :keyword client: Required. The schema registry client
      which is used to register schema and retrieve schema from the service.
-    :type client: ~azure.schemaregistry.SchemaRegistryClient
-    :param str group_name: Schema group under which schema should be registered.
+    :paramtype client: ~azure.schemaregistry.SchemaRegistryClient
+    :keyword str group_name: Required. Schema group under which schema should be registered.
     :keyword bool auto_register_schemas: When true, register new schemas passed to serialize.
      Otherwise, and by default, fail if it has not been pre-registered in the registry.
 
     """
 
-    def __init__(self, client, group_name, **kwargs):
-        # type: ("SchemaRegistryClient", str, Any) -> None
-        self._schema_group = group_name
+    def __init__(self, **kwargs):
+        # type: (Any) -> None
+        try:
+            self._schema_group = kwargs.pop("group_name")
+            self._schema_registry_client = kwargs.pop("client") # type: "SchemaRegistryClient"
+        except KeyError as e:
+            raise ValueError("'{}' is a required keyword.".format(e.args[0]))
```

Comment:
I think it should be a TypeError instead of ValueError.

When we have positional argument in `__init__` and we don't pass the argument, we would get a TypeError. I think we want the same thing here.

refs:
https://docs.python.org/3/library/exceptions.html#TypeError

Reply:
ah yep, makes sense